### PR TITLE
module3

### PR DESCRIPTION
**Pull request by AlexandraRyzvanovich**

### CODE REVIEW

File: src/Index.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import App from "./components/app";
+import Footer from "./components/common/footer";
+import Header from "./components/common/header";
 
 ReactDOM.render(<App />, document.getElementById("root"));
+ReactDOM.render(<Header />, document.getElementById("header"));
```
Comment (sushchenja) [inline src/Index.js:8]: use one entry point

File: src/components/app.js [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
-    render() {
-        return (
-            <div>
-                <h1>My React App!</h1>
-            </div>
-        );
-    }
+  render() {
+    return (
+      <Fragment>
```
Comment (sushchenja) [inline src/components/app.js:11]: unnecessary fragment

File: src/components/movieSorter.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            <button>
+              <p>SORT BY</p>
+            </button>
+            <button>
+              <p>RELEASE DATE</p>
+            </button>
+          </div>
+        </div>
+        <div className="line" />
+      </Fragment>
```
Comment (sushchenja) [inline src/components/movieSorter.js:38]: unnecessary fragment. to render the line pls use a pseudo element

File: src/components/movieSorter.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Component } from "react";
+import { Fragment } from "react";
+import MovieService from "../service/movieService";
+import "./../styles/movieListHeader.scss";
+import MovieList from "./movieList";
+
+class MovieSorter extends Component {
+  getAllMovies() {
```
Comment (sushchenja) [inline src/components/movieSorter.js:9]: what is the purpose of this method

File: src/components/movieList.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import React, { Component } from "react";
+import "./../styles/movieList.scss";
+import { Fragment } from "react";
+import PropTypes from "prop-types";
+import MovieService from "../service/movieService";
+
+class MovieList extends Component {
+  static propTypes = {
+    list: PropTypes.any,
```
Comment (sushchenja) [inline src/components/movieList.js:9]: don't use any

File: src/components/movieList.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  };
+  getDefaultList() {
+    return MovieService.getAllMovies();
+  }
+
+  render() {
+    return (
+      <Fragment>
+        <p className="list-size-text">
+          {this.propTypes?.list?.size || this.getDefaultList().size} MOVIES
```
Comment (sushchenja) [inline src/components/movieList.js:24]: you can't use propTypes like this

File: src/components/movieList.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+class MovieList extends Component {
+  static propTypes = {
+    list: PropTypes.any,
+  };
+
+  movies = {
+    size: 0,
+    items: [],
+  };
+  getDefaultList() {
```
Comment (sushchenja) [inline src/components/movieList.js:16]: let's try to implement a container component which will provide all necessary data. and leave this component as presentational

File: src/components/movieList.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    return (
+      <Fragment>
+        <p className="list-size-text">
+          {this.propTypes?.list?.size || this.getDefaultList().size} MOVIES
+          FOUND
+        </p>
+        <div className="list-wrapper">
+          {this.propTypes?.list?.items || this.getDefaultList().items}
+        </div>
+        ;
```
Comment (sushchenja) [inline src/components/movieList.js:30]: semicolon

File: src/service/movieService.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+      img: "https://www.kino-teatr.ru/movie/posters/big/1/32751.jpg",
+      name: "Lion King",
+      type: "Cartoon",
+      year: 1994,
+    });
+    debugger;
+    let movie;
+    moviesList.forEach((item, index) => {
+      movie = (
+        <Movie
```
Comment (sushchenja) [inline src/service/movieService.js:52]: service should provide only pure data

File: src/service/movieService.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
+import React from "react";
```
Comment (sushchenja) [inline src/service/movieService.js:1]: there is no place for react and component imports

File: src/components/movieList.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+  render() {
+    return (
+      <Fragment>
+        <p className="list-size-text">
+          {this.propTypes?.list?.size || this.getDefaultList().size} MOVIES
+          FOUND
+        </p>
+        <div className="list-wrapper">
+          {this.propTypes?.list?.items || this.getDefaultList().items}
```
Comment (sushchenja) [inline src/components/movieList.js:28]: get data from service, pass it to the view component and render it there

**Review by sushchenja — CHANGES_REQUESTED** [review]

File: src/components/movieComponent.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+    let showEdit = this.state.isEditOpened;
+    this.setState({
+      isEditOpened: !showEdit,
+    });
+  };
+
+  toggleDelete = () => {
+    let showDelete = this.state.isDeleteOpened;
+    this.setState({
+      isDeleteOpened: !showDelete,
```
Comment (sushchenja) [inline src/components/movieComponent.js:27]: you can use functional setState

**Review by sushchenja — CHANGES_REQUESTED** [review]

File: src/components/addPopup.js [hunk 13]
```diff
@@ -3,7 +3,7 @@ import { Component } from "react";
 
 import "./../styles/header.scss";
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/components/addPopup.js]: fixed

File: src/styles/movie.scss [hunk 14]
```diff
@@ -72,33 +72,128 @@ img {
   background-color: #020e03;
 }
 
+// .dropdown-content {
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/styles/movie.scss]: fixed

File: src/components/movieAppContainer.js [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+
+  handleOpenHeader = (id) => {
+    this.setState({
+      movieId: id,
+    });
+    this.toggleHeader();
+  };
+
+  toggleHeader = () => {
+    let isDefaultHeaderOpened = this.state.isDefaultHeaderOpened;
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/components/movieAppContainer.js]: fixed

File: src/components/movieAppContainer.js [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+    let isMovieInfoOpened = this.state.isMovieInfoOpened;
+    this.setState({
+      isDefaultHeaderOpened: !isDefaultHeaderOpened,
+      isMovieInfoOpened: !isMovieInfoOpened,
+    });
+  };
+
+  render() {
+    return (
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/components/movieAppContainer.js:70]: fixed

File: src/components/movieAppContainer.js [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+            onSearch={this.handleSearch}
+          />
+        )}
+        <MovieListContainer
+          onGetFilms={this.handleGetFilms}
+          items={this.state.films}
+          onGetMovieInfo={this.handleOpenHeader}
+        />
+        <AddPopup
+          isOpened={this.state.isOpened}
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/components/movieAppContainer.js]: fixed

File: src/components/movieListComponent.js [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+      );
+      list.push(movie);
+    });
+    return list;
+  };
+
+  const movies = useCallback(callback, deps);
+
+  return (
+    <Fragment>
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/components/movieListComponent.js]: fixed

File: src/components/movieListComponent.js [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+    let list = [];
+    let movie;
+    props.items.forEach((item, index) => {
+      movie = (
+        <MovieComponent
+          img={item.img}
+          name={item.name}
+          type={item.type}
+          year={item.year}
+          key={index + ""}
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/components/movieListComponent.js]: fixed

File: src/components/movieListComponent.js [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+          onGetMovieInfo={props.onGetMovieInfo}
+        />
+      );
+      list.push(movie);
+    });
+    return list;
+  };
+
+  const movies = useCallback(callback, deps);
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/components/movieListComponent.js]: fixed

File: src/components/movieInfoHeader.js [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+
+import "./../styles/infoHeader.scss";
+import "./../styles/common.scss";
+
+import LogoComponent from "./logoComponent.js";
+import MovieService from "../service/movieService";
+
+function MovieInfoHeader(props) {
+  let [movie, setMovie] = useState(() => {
+    let movie = MovieService.getFullMovieInfo(props.id);
```
[the remark this answers is not in the document]
Comment (AlexandraRyzvanovich) [reply src/components/movieInfoHeader.js]: fixed

File: src/components/movieComponent.js (hunk 12 above)
Comment (AlexandraRyzvanovich) [reply src/components/movieComponent.js]: fixed

File: src/components/editPopup.js [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+function EditPopup(props) {
+  const [movie, setMovie] = useState("");
+
+  useEffect(() => {
+    loadMovie();
+  }, []);
+
+  const loadMovie = () => {
+    setMovie(MovieService.getFullMovieInfo(1));
+  };
```
Comment (sushchenja) [inline src/components/editPopup.js:15]: all this line could be changed to custom hook: const movie = useMovie(1)

**Pull request merged**